### DISCUSSION TITLE
Confused with depth check in NewMagickWand func

### DISCUSSION THREAD

**Discussion by euph0nium** [ImageMagick/ImageMagick#4645] [Development] [answered]
Hello, I'm learning ImageMagick code. This code snipes confuse me. 
```c
/* magick-wand.c */
 depth=MAGICKCORE_QUANTUM_DEPTH;
 quantum=GetMagickQuantumDepth(&depth);
 if (depth != MAGICKCORE_QUANTUM_DEPTH)
    ThrowWandFatalException(WandError,"QuantumDepthMismatch",quantum);
```

```c
/* version.c */
MagickExport const char *GetMagickQuantumDepth(size_t *depth)
{
  if (depth != (size_t *) NULL)
    *depth=(size_t) MAGICKCORE_QUANTUM_DEPTH;
  return(MagickQuantumDepth);
}
```
Variable depth has been assigned to MAGICKCORE_QUANTUM_DEPTH.  When will depth different from MAGICKCORE_QUANTUM_DEPTH ?

**Comment by dlemstra** [comment, marked as answer]
It is possible that `depth` has a different value when you are mixing two versions of ImageMagick that have a different `MAGICKCORE_QUANTUM_DEPTH`. An example would be that you build `MagickCore` with `Q8` and `MagickWand` with `Q16`. Mixing those would result in a lot of issues and this check inside `NewDrawingWand` is there to make sure that you are not mixing different quantum depths. 

We now have this check in a couple places but I do wonder if we need them everywhere. The `NewPixelIterator` method requires a `MagickWand` that already has this check. And I also wonder if we should not create a new private method that we use inside `NewDrawingWand`, `NewMagickWand`, `NewPixelWand` and `AcquireMagickCLI` that has a name that makes it more clear why we are doing these checks.
